pkg/apis/ark/v1/restore.go

```diff
@@ -103,6 +107,9 @@ type RestoreStatus struct {
 	// Errors is a count of all error messages that were generated during
 	// execution of the restore. The actual errors are stored in object storage.
 	Errors int `json:"errors"`
+
+	// FailureReason is an error that caused the entire restore to fail.
+	FailureReason string `json:"failureReason"`
```

Comment:
I understand the motivation for this, but I do worry about the fact that we're adding yet another place to look for restore errors (ark server logs, restore logs, restore warnings/errors file, and now this field).   Undecided right now on whether we need to do something about it for this PR, or if it can hold off 'til we address that holistically.

Reply:
This is a good point. Per Slack convo, most of the failure cases should be in the restore log today; they'd just be marked as `Completed`. The only one where it would fail and it's not logged is if the upload of the log itself fails.

Reply:
Eh, looking at this again I think I'm fine with it.  Right now, this will only be populated if the restore doesn't even start, in which case it won't have a log or results in object storage.